### PR TITLE
Added LayoutFormatterPreferences (and related files)

### PR DESCRIPTION
**Pull request by oscargus**

### CODE REVIEW

File: src/main/java/net/sf/jabref/gui/BasePanel.java [hunk 1]
```diff
@@ -990,7 +991,8 @@ private void copyKeyAndTitle() {
                     "\\bibtexkey - \\begin{title}\\format[RemoveBrackets]{\\title}\\end{title}\n");
             Layout layout;
             try {
-                layout = new LayoutHelper(sr, Globals.prefs, Globals.journalAbbreviationLoader).getLayoutFromText();
+                layout = new LayoutHelper(sr, LayoutFormatterPreferences.fromPreferences(Globals.prefs),
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/gui/BasePanel.java]: I would also include the `journalAbbreviationLoader` in `LayoutFormatterPreferences`.

File: src/main/java/net/sf/jabref/gui/openoffice/OpenOfficePanel.java [hunk 2]
```diff
@@ -126,8 +127,8 @@ private OpenOfficePanel() {
         update = new JButton(IconTheme.JabRefIcon.REFRESH.getSmallIcon());
         update.setToolTipText(Localization.lang("Sync OpenOffice/LibreOffice bibliography"));
         preferences = new OpenOfficePreferences(Globals.prefs);
-        loader = new StyleLoader(preferences, Globals.prefs, Globals.journalAbbreviationLoader,
-                Globals.prefs.getDefaultEncoding());
+        loader = new StyleLoader(preferences, LayoutFormatterPreferences.fromPreferences(Globals.prefs),
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/gui/openoffice/OpenOfficePanel.java]: So if the OOPanel is open and I make some changes to the preferences, then `StyleLoader` is not changed, i.e. every action from the OOPanel still uses the old preferences, right?

**Comment by tobiasdiez** [comment]
LGTM :+1:

File: src/main/java/net/sf/jabref/gui/openoffice/OpenOfficePanel.java (hunk 2 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/openoffice/OpenOfficePanel.java]: Currently, yes.

File: src/main/java/net/sf/jabref/gui/BasePanel.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/BasePanel.java]: Yes, that may make sense. I made an attempt including in JournalAbbreviationPreferences, but that wasn't the correct place it turned out. Will try again.

**Comment by oscargus** [comment]
I included `JournalAbbreviationLoader` in `LayoutPreferences`. Tests failing because of non-related connection issues.

**Pull request merged**